dev/src/DocFx/Node/XrefTrait.php

```diff
@@ -87,20 +87,20 @@ function ($matches) {
     private function replaceProtoRef(string $description): string
     {
         return preg_replace_callback(
-            '/\[([^\]]*?)\]\s?\[([a-z1-9\.]*)([a-zA-Z1-9_\.]*)\]/',
+            '/\[([^\]]*?)\]\s?\[([a-z1-9\._]*)([a-zA-Z1-9_\.]*)\]/',
             function ($matches) {
                 list($link, $name, $package, $class) = $matches;
                 $property = $method = $constant = null;
 
                 // if the last word is all lowercase, it's a property
                 // if the last word is all uppercase, it's a constant
                 // otherwise, it's a nested class
-                if (preg_match('/([a-zA-Z\.]+)?\.([a-z_]+)$/', $class, $matches)) {
-                    $class = $matches[1];
-                    $property = $matches[2];
-                } elseif (preg_match('/([a-zA-Z\.]+)?\.([A-Z_]+)$/', $class, $matches)) {
-                    $class = $matches[1];
-                    $constant = $matches[2];
+                if (preg_match('/([a-zA-Z\.]+)?\.([a-z_1-9]+)$/', $class, $propertyMatches)) {
```

Comment:
Where does this `$propertyMatches` variable comes from?

Reply:
After reading the documentation of `preg_match` I can see that the 3rd parameter receives a pointer to a variable.

Is this creating a new property that then we use inside the `if` logic?

Reply:
yes! exactly. We're providing a variable for it to populate, if it's successful.

This only changes it from using `$matches` (which we populated above) to using its own specific variable `$propertyMatches`. Otherwise, the previous values in `$matches` were being overwritten, which was both confusing and also not a good practice.

Reply:
Got it! Thanks for the clarification!